### PR TITLE
feat: set up table package and add createTableItems method

### PR DESCRIPTION
**Pull request by square-li**
- config typescript & rollup
- config jest
- config eslint
- add unit tests for createTableItems method
- updated .eslintrc.js at project root level

## Overview
Set up new table package. Added createTableItems method.

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by diehbria — CHANGES_REQUESTED** [review]

File: packages/table/src/utils/createTableItems.spec.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    invalid: {
+      $cellRef: {
+        id: 'invalid-data-stream',
+        resolution: 0,
+      },
+    },
+  },
+];
+
+describe('createTableItems method', () => {
```
[the remark this answers is not in the document]
Comment (square-li) [reply packages/table/src/utils/createTableItems.spec.ts:64]: Yes removing it.

File: packages/table/src/utils/createTableItems.spec.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  it('should create table items', () => {
```
[the remark this answers is not in the document]
Comment (square-li) [reply packages/table/src/utils/createTableItems.spec.ts]: Updated.

File: packages/table/src/utils/dataFilter.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { bisector } from 'd3-array';
```
[the remark this answers is not in the document]
Comment (square-li) [reply packages/table/src/utils/dataFilter.ts]: Copied.

File: packages/table/src/utils/createTableItems.spec.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    invalidAggregation: {
+      $cellRef: {
+        id: 'agg_data',
+        resolution: 55,
+      },
+    },
+  },
+];
+
+describe('createTableItems', () => {
```
Comment (diehbria) [inline packages/table/src/utils/createTableItems.spec.ts:84]: still have unnecessary describe blocks

Comment (square-li) [reply packages/table/src/utils/createTableItems.spec.ts:84]: I see what do you mean now. Removing all describe blocks.

File: packages/table/src/utils/types.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+export type CellItemProps = {
+  value?: Primitive;
+  error?: ErrorDetails;
+  isLoading?: boolean;
+  threshold?: Threshold;
+};
+
+export class CellItem {
```
Comment (diehbria) [inline packages/table/src/utils/types.ts:22]: A class is not a type, unsure if this is the right place for this. is there a reason to use a class for this?

File: packages/table/src/utils/createTableItems.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { Viewport, DataStream } from '@iot-app-kit/core';
+import { breachedThreshold, Primitive, Threshold, DataStream as SC_DataStream } from '@synchro-charts/core';
```
Comment (diehbria) [inline packages/table/src/utils/createTableItems.ts:2]: we refer to it as `SynchroChartsDataStream` else where in IoT App Kit

**Review by diehbria — APPROVED** [review]

**Pull request merged**